gallery_dl/extractor/soundgasm.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extractors for https://soundgasm.net/"""
+
+from .common import Extractor, Message
+from .. import text
+
+
+class SoundgasmSoundExtractor(Extractor):
+    """Extractor for individual sounds on soundgasm.net"""
+    category = "soundgasm"
+    subcategory = "sound"
+    filename_fmt = "{title}.{extension}"
+    directory_fmt = ("{category}", "{user}")
+    archive_fmt = "{user}_{slug}"
+    pattern = (r"(?:https?://)?(?:www\.)?soundgasm\.net"
+               r"/u(?:ser)?/([^/?#]+)/([^/?#]+)")
+    test = (
+        (("https://soundgasm.net/user/fierce-aphrodite"
+         "/Soap-by-Melanie-Martinez-Cover"), {
+            "pattern": (r"https://media\.soundgasm\.net/sounds"
+                        r"/04614d5a142ee4d56967ed2b764de3453d048c5b\.m4a"),
+            "keyword": {
+                "user" : "fierce-aphrodite",
+                "slug" : "Soap-by-Melanie-Martinez-Cover",
+                "title": "Soap by Melanie Martinez (Cover)",
+                "description": str,
+            },
+        }),
+        ("https://soundgasm.net/u/anner1234/Your-girlfriend-is-sick", {
+            "pattern": (r"https://media\.soundgasm\.net/sounds"
+                        r"/db4d34bb9e11f71385fce1fb6bd17674faaae483\.m4a"),
+            "keyword": {
+                "user" : "anner1234",
+                "slug" : "Your-girlfriend-is-sick",
+                "title": "Your girlfriend is sick",
+                "description": str,
+            },
+        }),
+    )
+
+    def __init__(self, match):
+        Extractor.__init__(self, match)
+        self.user, self.slug = match.groups()
+
+    def items(self):
+        page = self.request("https://soundgasm.net/user/{}/{}".format(
+            self.user, self.slug)).text
+        data = {
+            "user" : self.user,
+            "slug" : self.slug,
+            "title": text.unescape(text.extr(
+                page, 'aria-label="title">', "</div>")),
+            "description": text.unescape(text.remove_html(text.extr(
+                page, 'class="jp-description">', "</div>"))),
+        }
+        url = text.extract(page, ': "', '"', page.index('("setMedia",'))[0]
```

Comment:
```suggestion
        url = text.extr(page, 'm4a: "', '"')
```
should i do this?
I don't know if there can be a different `ext` all the links I came across had `m4a` in this snippet
```
$(this).jPlayer("setMedia", {
    m4a: "https://media.soundgasm.net/sounds/04614d5a142ee4d56967ed2b764de3453d048c5b.m4a"
});
```

Reply:
Yeah, this would obviously break for any other `ext`, so I had solved this by extracting all potential formats and just picking the first one.